build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/BwcSetupExtension.java

```diff
@@ -65,11 +66,14 @@ public BwcSetupExtension(
             loggedExec.usesService(bwcTaskThrottleProvider);
             loggedExec.setSpoolOutput(true);
             loggedExec.setWorkingDir(checkoutDir.get());
-            loggedExec.doFirst(t -> {
-                // Execution time so that the checkouts are available
-                String compilerVersionInfoPath = minimumCompilerVersionPath(unreleasedVersionInfo.get().version);
-                String minimumCompilerVersion = readFromFile(new File(checkoutDir.get(), compilerVersionInfoPath));
-                loggedExec.environment("JAVA_HOME", getJavaHome(Integer.parseInt(minimumCompilerVersion)));
+            loggedExec.doFirst(new Action<Task>() {
```

Comment:
lambdas as task actions are now deprecated. therefore we do not need a comment here as the build would fail to signal this anyhow